source/common/http/utility.cc

```diff
@@ -306,8 +306,32 @@ void Utility::sendLocalReply(
     std::function<void(Buffer::Instance& data, bool end_stream)> encode_data, const bool& is_reset,
     Code response_code, absl::string_view body_text,
     const absl::optional<Grpc::Status::GrpcStatus> grpc_status, bool is_head_request) {
+  sendLocalReply(
+      is_grpc, encode_headers, encode_data, [&](Code&) -> void {},
+      [&](HeaderMapPtr&& response_headers, absl::string_view& body) -> std::string {
+        if (!body.empty()) {
+          response_headers->insertContentLength().value(body.size());
+          response_headers->insertContentType().value(Headers::get().ContentTypeValues.Text);
+          return body.data();
+        }
+        return {};
+      },
+      is_reset, response_code, body_text, grpc_status, is_head_request);
+}
+
+void Utility::sendLocalReply(
+    bool is_grpc, std::function<void(HeaderMapPtr&& headers, bool end_stream)> encode_headers,
```

Comment:
This method signature keeps growing and growing :) Can we wrap the parameters up into one or more structs that group things together?